tasks/rosetta-code/count-examples/src/lib.rs

```diff
@@ -0,0 +1,112 @@
+extern crate reqwest;
+extern crate url;
+extern crate rustc_serialize;
+
+use std::io::Read;
+use self::url::Url;
+use rustc_serialize::json::{self, Json};
+
+pub struct Task {
+    page_id: u64,
+    pub title: String,
+}
+
+#[derive(Debug)]
+enum ParseError {
+    /// Something went wrong with the HTTP request to the API.
+    Http(reqwest::Error),
+
+    /// There was a problem parsing the API response into JSON.
+    Json(json::ParserError),
+
+    /// Unexpected JSON format from response
+    UnexpectedFormat,
+}
+impl From<json::ParserError> for ParseError {
+    fn from(error: json::ParserError) -> Self {
+        ParseError::Json(error)
+    }
+}
+
+impl From<reqwest::Error> for ParseError {
+    fn from(error: reqwest::Error) -> Self {
+        ParseError::Http(error)
+    }
+}
+
+
+fn construct_query_category(category: &str) -> Url {
+    let mut base_url = Url::parse("http://rosettacode.org/mw/api.php").unwrap();
+    let cat = format!("Category:{}", category);
+    let query_pairs = vec![("action", "query"),
+                           ("format", "json"),
+                           ("list", "categorymembers"),
+                           ("cmlimit", "500"),
+                           ("cmtitle", &cat),
+                           ("continue", "")];
+    base_url.query_pairs_mut().extend_pairs(query_pairs.into_iter());
+    base_url
+}
+
+fn construct_query_task_content(task_id: &str) -> Url {
+    let mut base_url = Url::parse("http://rosettacode.org/mw/api.php").unwrap();
+    let mut query_pairs =
+        vec![("action", "query"), ("format", "json"), ("prop", "revisions"), ("rvprop", "content")];
+    query_pairs.push(("pageids", task_id));
+    base_url.query_pairs_mut().extend_pairs(query_pairs.into_iter());
+    base_url
+}
+
+fn query_api(url: Url) -> Result<Json, ParseError> {
+    let mut response = try!(reqwest::get(url.as_str()));
+    // Build JSON
+    let mut body = String::new();
+    response.read_to_string(&mut body).unwrap();
```

Comment:
So I have been looking into it and it looks like the purpose of that function would be to deserialize the entire response similar to how they do it [here](https://github.com/serde-rs/json). Do you think it would be good idea then to create a structure for the two types of responses and then use the derivation of deserialize as serde seem to propse?